lib/src/rules/avoid_types_on_closure_parameters.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.avoid_annotating_types_on_function_expressions;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Avoid annotating types on function expression parameters.';
```

Comment:
@bwilkerson what do you think about the message?.

The issues were fixed, 114 lints in the SDK no false positives and no more than 1 lint per method.